workflow/scripts/modify_prenetwork.py

```diff
@@ -234,6 +234,66 @@ def add_wasserstoff_kernnetz(n, wkn, costs):
 
     # from 2030 onwards all pipes are extendable (except from the ones the model build up before and the kernnetz lines)
 
+def unravel_oilbus(n):
+    """
+    Unravel European oil bus to enable energy balances for import of oil products.
+
+    """
+    logger.info("Unraveling oil bus")
+    # add buses
+    n.add("Bus", "DE oil", carrier="oil")
+    n.add("Bus", "DE renewable oil", carrier="e-fuels")
+    n.add("Bus", "EU renewable oil", carrier="e-fuels")
+
+    # add one generator for DE oil
+    n.add("Generator",
+          name="DE oil",
+          bus="DE oil",
+          carrier="fossil oil import",
+          p_nom_extendable=True,
+          marginal_cost=n.generators.loc["EU oil"].marginal_cost,
+          )
+    
+    # change links from EU oil to DE oil
+    german_oil_links = n.links[(n.links.bus0=="EU oil") & (n.links.index.str.contains("DE"))].index
+    german_FT_links = n.links[(n.links.bus1=="EU oil") & (n.links.index.str.contains("DE"))].index
+    n.links.loc[german_oil_links, "bus0"] = "DE oil"
+    n.links.loc[german_FT_links, "bus1"] = "DE renewable oil"
+
+    # change FT links in rest of Europe
+    europ_FT_links = n.links[n.links.bus1=="EU oil"].index
+    n.links.loc[europ_FT_links, "bus1"] = "EU renewable oil"
+
+    # add links between oil buses
+    n.madd(
+        "Link",
+        ["EU renewable oil -> DE oil", "EU renewable oil -> EU oil", "DE renewable oil -> DE oil", "DE renewable oil -> EU oil"],
+        bus0=["EU renewable oil", "EU renewable oil", "DE renewable oil", "DE renewable oil"],
+        bus1=["DE renewable oil", "EU oil", "DE oil", "EU oil"],
+        carrier="e-fuels",
```

Comment:
I am thinking now if the carrier shouldn't be "Fischer-Tropsch" instead of e-fuels? at the moment we have three different names to refer to Fischer-Tropsch products:

"Fischer-Tropsch" for the plants, "renewable oil" for the buses, and "e-fuels" for the trade links